pep8.py

```diff
@@ -1846,7 +1846,15 @@ def check_files(self, paths=None):
         report.start()
         try:
             for path in paths:
-                if os.path.isdir(path):
+                if path == "-":
+                    # STDIN
+                    path = self.options.stdin_name
```

Comment:
The explanation of the other question above shows exactly what happens when someone doesn't specify a name.

Yes, `path` will be `None`, but `Checker` is expecting it and will assume the file is name `stdin`.